packages/components/Badge/src/Badge.tsx

```diff
@@ -44,27 +43,33 @@ export const Badge = compose<BadgeType>({
     const Slots = useSlots(userProps, (layer) => badgeLookup(layer, userProps));
 
     return (final: BadgeProps, ...children: ReactNode[]) => {
-      const { icon, iconPosition, size, ...mergedProps } = mergeProps(badge, final);
+      const { icon, iconPosition, size, appearance, ...mergedProps } = mergeProps(badge, final);
       const showContent = size !== 'tiny' && size !== 'extraSmall';
       const showIcon = size !== 'tiny';
-      return (
-        <Slots.shadow>
-          <Slots.root {...mergedProps}>
-            {icon && showIcon && iconPosition === 'before' && <Slots.icon accessible={false} {...iconProps} />}
-            {showContent &&
-              Children.map(children, (child, i) =>
-                typeof child === 'string' ? (
-                  <Slots.text accessible={false} key={`text-${i}`}>
-                    {child}
-                  </Slots.text>
-                ) : (
-                  child
-                ),
-              )}
-            {icon && showIcon && iconPosition === 'after' && <Slots.icon accessible={false} {...iconProps} />}
-          </Slots.root>
-        </Slots.shadow>
+      const showOptionalShadow = appearance === 'filled' || appearance === 'tint';
+
+      const badgeWithoutShadow = (
+        <Slots.root {...mergedProps}>
+          {icon && showIcon && iconPosition === 'before' && <Slots.icon accessible={false} {...iconProps} />}
+          {showContent &&
+            Children.map(children, (child, i) =>
+              typeof child === 'string' ? (
+                <Slots.text accessible={false} key={`text-${i}`}>
+                  {child}
+                </Slots.text>
+              ) : (
+                child
+              ),
+            )}
+          {icon && showIcon && iconPosition === 'after' && <Slots.icon accessible={false} {...iconProps} />}
+        </Slots.root>
       );
+
+      if (showOptionalShadow) {
+        return <Slots.shadow>{badgeWithoutShadow}</Slots.shadow>;
+      } else {
+        return badgeWithoutShadow;
```

Comment:
`Shadow.tsx` will only render the children if no shadow token is applied
```
  if (!props.shadowToken) {
    return <>{children}</>;
  }
```
Could we do something where we only pass the shadow Token if our appearance is `filled` or `tint` so that we can just take advantage of the existing optimization in Shadow?

Reply:
Yeah I'm guessing that's how the code was meant to work based on what was written - always use the shadow slot but only apply shadow tokens if the appearance is one of the correct ones.

Reply:
Hmm so I had briefly looked into this but wasn't sure where the logic to "only pass the shadow Token if our appearance is filled or tint" would go.

In Notification we could define the shadow tokens in NotificationTokens, because by default Shadows should always be applied to certain Notification variants. For badge the shadow is only optional for certain badge variants (by default it's not applied), so there aren't any shadows defined in BadgeTokens. The way to add a shadow in Badge is by using Badge.customize

Maybe if a user really wants to use customize to add a shadow to a badge variant that shouldn't have a shadow, we could just let them? And maybe just print out a warning.

Reply:
Talked offline - decided we should leave it up the user to not add a shadow to certain variants. I've reverted most of the changes in the badge component, and added a warning in the tester.